eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -346,19 +352,42 @@ def insert_struct_to_complex(self, struct, complex):
         """
         for i in range(len(complex.children)):
             if self.get_name(struct) == self.get_name(complex.children[i].children[0]):
-                complex.children[i] = Tree('agent', [struct])
-                break
-        return complex
+                struct_found = True
+                for j in range(len(struct.children[1].children)):
+                    for k in range(
+                        len(complex.children[i].children[0].children[1].children)
+                    ):
+                        if self.get_name(
+                            struct.children[1].children[j]
+                        ) == self.get_name(
+                            complex.children[i].children[0].children[1].children[k]
+                        ):
+                            struct_found = False
+                            break
+                    if not struct_found:
+                        break
```

Comment:
Please explain (or, even better, add comments to docstring of this function), what is the purpose of this part. Might be useful to clarify that we are looking at atomics and why.

Reply:
I added comments to `insert_struct_to_complex` and expanded docstring with details for each of the function.